src/test/integration/accounts.e2e-spec.ts

```diff
@@ -1,226 +1,179 @@
-
-        expect(accountDetailed.balance).toBeDefined();
-        expect(Number(accountDetailed.balance)).toBeGreaterThan(0);
-      });
+      for (const result of results) {
+        expect(results).toHaveLength(10);
+        expect(result).toHaveStructure(Object.keys(new Account()));
+      }
+    });
 
-      it(`should throw 'Account not found' error`, async () => {
-        expect(await accountService.getAccount(userAccount.address + 'a')).toBeNull();
-      });
+    it("returned accounts contains properties", async () => {
+      jest
+        .spyOn(CachingService.prototype, 'getOrSetCache')
+        // eslint-disable-next-line require-await
+        .mockImplementation(jest.fn(async (_key: string, promise: any) => promise()));
 
-    });
+      const results = await accountService.getAccounts({ from: 0, size: 10 });
 
-    describe('Deferred Account', () => {
-      it(`should return a list of deferred accounts`, async () => {
-        const account = await accountService.getDeferredAccount(userAccount.address);
-        expect(account).toBeInstanceOf(Array);
-      });
+      for (const result of results) {
+        expect(result.hasOwnProperty("address")).toBeTruthy();
+        expect(result.hasOwnProperty("balance")).toBeTruthy();
+        expect(result.hasOwnProperty("nonce")).toBeTruthy();
+        expect(result.hasOwnProperty("shard")).toBeTruthy();
+      }
     });
+  });
 
-    describe('Account Delegation Legacy', () => {
-      it(`should return a delegation legacy for an account with address`, async () => {
-        const account = await delegationLegacyService.getDelegationForAddress(userAccount.address);
-        expect(account).toHaveStructure(Object.keys(new AccountDelegationLegacy()));
-      });
-    });
+  describe("Get Accounts Contract", () => {
+    it("should return contracts accounts", async () => {
+      const contractAddress: string = "erd1ss6u80ruas2phpmr82r42xnkd6rxy40g9jl69frppl4qez9w2jpsqj8x97";
+      const results = await accountService.getAccountContracts({ from: 0, size: 5 }, contractAddress);
 
-    describe('Account username based on Address', () => {
-      it('should return account username based on address ', async () => {
-        const username = await accountService.getAccountUsername(userAccount.address);
-        expect(username).toBe(userAccount.username);
-      });
+      for (const result of results) {
+        expect(results).toHaveLength(5);
+        expect(result).toHaveStructure(Object.keys(new DeployedContract()));
+      }
     });
+  });
 
-    describe('Account Deployed', () => {
-      it(`should return the deployed timestamp for a given address`, async () => {
-        const accountDeployed = await accountService.getAccountDeployedAt(providerAccount.address);
-        expect(typeof accountDeployed).toBe('number');
-      });
-    });
+  describe("Get Account Username", () => {
+    it("should return account username", async () => {
+      jest
```

Comment:
Please move this mock implementation in beforeAll function

Reply:
Fixed